logger.js

```diff
@@ -0,0 +1,205 @@
+const levels = {
+  info: "INFO",
+  warn: "WARN",
+  error: "ERROR",
+  debug: "DEBUG"
+}
+
+var currentLogLevel = "info"
+let isLoggerEnabled = true
+
+function getTimeStamp(){
+  const date = new Date();
+  return date.getFullYear() + "-" + (date.getMonth() + 1) + "-" + date.getDate() +
+      " " + date.getHours() + ":" + date.getMinutes() + ":" + date.getSeconds()
+}
+
+function log(level, message, details) {
+  if(!isLoggerEnabled) return;
+  let timeStamp = getTimeStamp();
+  let logEntry = `[${timeStamp}] [${level}] - ${message}`
+  if (details) { logEntry += " - Details: " + JSON.stringify(details) }
+  console.log(logEntry)
+  saveLogToFile(logEntry)
+}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Missing log level filtering implementation</b></div>
<div id="fix">

The `log` function doesn't respect the `currentLogLevel` setting. Consider implementing level-based filtering.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -17,6 +17,12 @@
 function log(level, message, details) {
   if(!isLoggerEnabled) return;
 +  const levels = {
 +    DEBUG: 0, INFO: 1, WARN: 2, ERROR: 3
 +  };
 +  if (levels[level] < levels[currentLogLevel.toUpperCase()]) {
 +    return;
 +  }
   let timeStamp = getTimeStamp();
   let logEntry = `[${timeStamp}] [${level}] - ${message}`
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/BitoPythonTest/pull/21#issuecomment-2456310783>#f099f2</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged